File: app/src/main/java/com/webviewblock/presentation/home/router/HomeNavigator.java
package com.webviewblock.presentation.home.router;

import com.webviewblock.R;
import com.webviewblock.navigator.NavigationController;
import com.webviewblock.navigator.NavigationEventData;
import com.webviewblock.navigator.Navigator;
import com.webviewblock.navigator.SharedEvents;

import javax.inject.Inject;

import timber.log.Timber;

public class HomeNavigator extends Navigator {

    @Inject
    public HomeNavigator(SharedEvents events) {
        super(events);
    }

    @Override
    public boolean handleNavigationEvent(NavigationEventData navigationState, NavigationController navController) {
        Timber.d("HomeNavigator navigation destination: " + navigationState.getEvent().name());
        switch (navigationState.getEvent()) {
            case SEARCH:
                navigate(
                        R.id.searchFragment,
                        navController,
                        navigationState.getData()
                );
                break;
            case SETTINGS:
                navigate(
                        R.id.settingsFragment,
                        navController,
                        navigationState.getData()
                );
                break;
            default:
                Timber.d("MainNavigator unknow navigation destination: " + navigationState.getEvent().name());
                break;
        }
        return false;
    }
}

File: app/src/main/java/com/webviewblock/presentation/home/router/HomeNavigator.kt
package com.webviewblock.presentation.home.router

import com.webviewblock.R
import com.webviewblock.domain.enums.NavigationEvent
import com.webviewblock.navigator.NavigationController
import com.webviewblock.navigator.NavigationEventData
import com.webviewblock.navigator.Navigator
import com.webviewblock.navigator.SharedEvents
import timber.log.Timber

open class HomeNavigator(sharedEvents: SharedEvents) : Navigator(sharedEvents) {

    override fun handleNavigationEvent(
        navigationState: NavigationEventData,
        navController: NavigationController
    ): Boolean {
        Timber.d("HomeNavigator navigation destination: " + navigationState.event.name)
        when (navigationState.event) {
            NavigationEvent.SEARCH -> {
                navigate(
                    R.id.searchFragment,
                    navController,
                    navigationState.data
                )
            }
            NavigationEvent.SETTINGS -> {
                navigate(
                    R.id.settingsFragment,
                    navController,
                    navigationState.data
                )
            }
            NavigationEvent.WEBVIEW -> {
//                navigate(
//                        R.id.cardsFragment,
//                        navController,
//                        navigationState.data
//                )
            }
            else -> {
                Timber.d("MainNavigator unknow navigation destination: " + navigationState.event.name)
                return false
            }
        }
        return true
    }
}